---
layout: default
title: Research
permalink: /research/
---
<div class = "research-page">

<div class = 'row'>
    <div class="col-sm-3 col-xs-6 ">
        <img class = "img-fix" src="{{"/img/network.png" | prepend: site.imgurl }}" />
    </div>
    <div class="col-sm-3 col-xs-6 ">
        <img class = "img-fix" src="{{"/img/spatial_localization.jpg" | prepend: site.imgurl }}"  />
    </div>
    <div class="clearfix visible-xs"></div>
    <div class="col-sm-3 col-xs-6">
        <img class = "img-fix" src="{{"/img/count-vs-umi.png" | prepend: site.imgurl }}" >
    </div>
    <div class="col-sm-3 col-xs-6">
        <img class = "img-fix" src="{{"/img/tsne-example.png" | prepend: site.imgurl }}" />
    </div>
</div>

  <br>

<h3> Statistical methods for single cell genomics </h3>
<p>
        Recent advances in molecular biology and microfluidics have enabled us to individually sequence the molecular contents of thousands of single cells. These datasets promise to transform our understanding of cellular diversity, but they are incomplete. Firstly, they contain information for only a small fraction of molecules in a cell - necessitating new methods capable of handling extensive technical noise. Secondly, they contain only molecular information about the cell, and lack crucial metadata about a cell’s environment, lineage, and interactions, that are essential towards understanding and predicting its behavior.
</p>
<p>
        To address these challenges, we are applying powerful tools in statistical inference and machine-learning to single cell data. We are particularly interested in building methods that can integrate information across multiple data types, modalities, and experimental conditions. We hope that these tools will be valuable not only across multiple projects in the lab (see below), but also to the broader community.
</p>

  <h3> Integrated analysis of cellular decision-making </h3>
  <p>
        Cellular diversity increases dramatically during differentiation, as a single progenitor can give rise to a breathtaking diversity of cell types. How do progenitor cells choose their terminal fates? Through single cell analysis, we are exploring the intrinsic and extrinsic factors which drive cellular decision-making. Single cell RNA-seq enables powerful approaches to reconstruct developmental trajectories, but we are also building an integrated framework to understand how a cell’s spatial localization, epigenomic landscape, and parental lineage influence its behavior and fate. Through projects driven both within the lab, and collaboratively across NYC, we focus on the development of the mammalian immune and nervous systems.
  </p>

  <h3> Deconvolution of autoimmune disease </h3>
  <p>
        The immune system strikes a delicate balance between immunity and tolerance, enabled by diverse subsets of interdependent and intercommunicating cells. Abnormal immune responses lead to autoimmune disease, and this extensive cellular heterogeneity can obscure the subpopulations and molecular pathways that drive disease progression and treatment. Working with the Hospital for Special Surgery and NYU, we’ve established a pipeline for high-throughput single cell analysis of clinical samples from patients with autoimmune disease, and are leveraging these rich datasets to better understand disease etiology and pathogenesis, with the goal of identifying new biomarkers and therapeutic avenues. More generally, we hope to build new experimental and computational tools that will advance and democratize clinical applications for single cell genomics.
  </p>


</div>

